community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/LuceneIndex.java

```diff
@@ -101,6 +101,7 @@
     public void addNode( long entityId, String key, Object value )
     {
         assertValidKey( key );
+        assertValidValue( value );
```

Comment:
It looks like `value` can be of type `ValueContext` and so the actual property value would be `ValueContext#getCorrectValue()`. Shouldn't we `assertValidValue` on the thing inside `ValueContext`?